docs(readme): improve readme (#7)
sapphire-project/pieces: README.md

<div align="center">

![Sapphire Logo](https://cdn.skyra.pw/gh-assets/sapphire.png)

# @sapphire/pieces

> Sapphire's piece loader</h4>

Reviewer: ```suggestion
Sapphire's piece loader.</h4>
```
also where's the other h4 tag

[![npm](https://img.shields.io/npm/v/@sapphire/pieces?color=crimson&logo=npm&style=flat-square)](https://www.npmjs.com/package/@sapphire/pieces)
[![Depfu](https://badges.depfu.com/badges/30f62cdba0c569448b5f93d532dad8e4/count.svg)](https://depfu.com/github/sapphire-project/pieces?project_id=15121)

</div>

## Description

> Pieces make up the core of the [Sapphire Framework][]. In the design of [Sapphire Framework][] every command, event, precondition and others in the root "pieces". They are pieces of your code, that make up the whole.

Reviewer: If you're trying to suggest commands and such are pieces...
```suggestion
Pieces make up the core of the [Sapphire Framework][]. In the design of [Sapphire Framework][] your commands, events, preconditions and more are "pieces". They are pieces of your code, that make up the whole.
```

**Much of the code in this repository is based on the ideas in [klasa]. This project would not exist without it.**

## API Documentation

For the full API documentation please refer to the TypeDoc generated [documentation](https://sapphire-project.github.io/pieces).

## Buy us some doughnuts

> Sapphire Project is and always will be open source, even if we don't get donations. That being said, we know there are amazing people who may still want to donate just to show their appreciation. Thanks you very much in advance!

Reviewer 2: ```suggestion
Sapphire Project is and always will be open source, even if we don't get donations. That being said, we know there are amazing people who may still want to donate just to show their appreciation. Thank you very much in advance!
```

<!-- ALL-CONTRIBUTORS-LIST:END -->

This project follows the [all-contributors](https://github.com/all-contributors/all-contributors) specification. Contributions of any kind welcome!

> [sapphire framework]: https://github.com/sapphire-project/framework

Reviewer: Capitalise it
```suggestion
[Sapphire Framework]: https://github.com/sapphire-project/framework
```

Author: Prettier is going to lowercase it again

Reviewer: Curse prettier
